src/azure-cli/azure/cli/command_modules/vm/_actions.py

```diff
@@ -91,10 +92,13 @@ def _load_images_from_publisher(publisher):
                         'publisher': publisher,
                         'offer': o.name,
                         'sku': s.name,
-                        'version': i.name
+                        'version': i.name,
+                        'architecture': i.additional_properties.get("properties", {}).get("architecture", None) or ""
```

Comment:
I'd like to ask if `i.additional_properties` might be None?

Reply:
The type of i should be [`VirtualMachineImageResource`](https://docs.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.v2020_12_01.models.virtualmachineimageresource?view=azure-python) class, documentation describes `additional_properties` as required, and the type is `ExtendedLocation`.